src/Microsoft.AspNetCore.Mvc.Core/Controllers/ControllerActivatorProvider.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Controllers
+{
+    /// <summary>
+    /// Provides methods to create an MVC controller.
+    /// </summary>
+    public class ControllerActivatorProvider : IControllerActivatorProvider
+    {
+        private static readonly Func<Type, ObjectFactory> _createFactory = (type) => ActivatorUtilities.CreateFactory(type, Type.EmptyTypes);
+        private static readonly Action<ControllerContext, object> _dispose = Dispose;
+        private readonly Func<ControllerContext, object> _controllerActivatorCreate;
+        private readonly Action<ControllerContext, object> _controllerActivatorRelease;
+
+        public ControllerActivatorProvider(IEnumerable<IControllerActivator> activators)
```

Comment:
I actually removed the implementations. My plan was to leave the contract in to continue other systems that implemented it, but not carry forward our implementation. Alternatively, we can register the current one and check for an exact type match.